src/System.Net.Http/src/System/Net/Http/Managed/ManagedHandler.cs

```diff
@@ -0,0 +1,307 @@
+            get { return _preAuthenticate; }
+            set { CheckInUse(); _preAuthenticate = value; }
+        }
+
+        public bool UseDefaultCredentials
+        {
+            get { return _useDefaultCredentials; }
+            set { CheckInUse(); _useDefaultCredentials = value; }
+        }
+
+        public ICredentials Credentials
+        {
+            get { return _credentials; }
+            set { CheckInUse(); _credentials = value; }
+        }
+
+        public bool AllowAutoRedirect
+        {
+            get { return _allowAutoRedirect; }
+            set { CheckInUse(); _allowAutoRedirect = value; }
+        }
+
+        public int MaxAutomaticRedirections
+        {
+            get { return _maxAutomaticRedirections; }
+            set { CheckInUse(); _maxAutomaticRedirections = value; }
+        }
+
+        public int MaxConnectionsPerServer
+        {
+            get { return _maxConnectionsPerServer; }
+            set
+            {
+                CheckInUse();
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+
+                _maxConnectionsPerServer = value;
+            }
+        }
+
+        public int MaxResponseHeadersLength
+        {
+            get { return _maxResponseHeadersLength; }
+            set
+            {
+                CheckInUse();
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+
+                _maxResponseHeadersLength = value;
+            }
+        }
+
+        public X509CertificateCollection ClientCertificates
+        {
+            get
+            {
+                if (_clientCertificateOptions != ClientCertificateOption.Manual)
+                {
+                    throw new InvalidOperationException(SR.Format(SR.net_http_invalid_enable_first, nameof(ClientCertificateOptions), nameof(ClientCertificateOption.Manual)));
+                }
+
+                return _clientCertificates ?? (_clientCertificates = new X509Certificate2Collection());
+            }
+        }
+
+        public Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool> ServerCertificateCustomValidationCallback
+        {
+            get { return _serverCertificateCustomValidationCallback; }
+            set { CheckInUse(); _serverCertificateCustomValidationCallback = value; }
+        }
+
+        public bool CheckCertificateRevocationList
+        {
+            get { return _checkCertificateRevocationList; }
+            set { CheckInUse(); _checkCertificateRevocationList = value; }
+        }
+
+        public SslProtocols SslProtocols
+        {
+            get { return _sslProtocols; }
+            set
+            {
+                CheckInUse();
+#pragma warning disable 0618 // obsolete warning
+                if ((value & (SslProtocols.Ssl2 | SslProtocols.Ssl3)) != 0)
```

Comment:
@morganbr, I get that using SSL3.0 is insecure and provides no protection when someone puts a little bit of effort in to break that protection. If we are going to remove SSL3.0 from being used at all because it's not secure, why aren't we making HttpClientHandler HTTPS only as HTTP is even more insecure than SSL3.0. Whatever security reason you can give me for why we should disable SSL3.0 also applies to HTTP.